### ISSUE TITLE
Gitolite tests fail.

### ISSUE THREAD

**Issue by pumzie** [ekapusta/docker-alpine-gitolite#2]
Hello,
I'm executing the Gitolite tests. They come out with errors.
Here ist my setup:
- Starting the container as described.
- Logging in the Container with "docker exec -it #ID /bin/sh
- Creating the User "gittest" (in the container)
- Installing the Test as described (http://gitolite.com/gitolite/testing.html)
- Running the test, two of them fail (info-json.t, repo-specific-hooks.t)

To fix info-json.t:
- apk add g++
- apk add make
- apk add perl-dev
- PERL_MM_USE_DEFAULT=1
- cpan -f JSON

To fix repo-specific-hooks.t:
No idea. I'm not in Perl. Building my own alpine container leads to the same error. Building on Ubuntu, Debian, ... works fine

**Comment by garex** [comment]
@pumzie do you think tests-relative info should be inside this image?

May be you will create another one, that will inherit this image? For example Dockerfile.tests

**Regarding tests.** Hmm.. I dont' thought it needs to be tested. But not I think it good to have tests inside this environment.

**Comment by pumzie** [comment]
No. The tests should not be inside the Image! I've done it "on the fly" to run the tests. The tests do not run remotely, but only locally. I've implanted them to run at runtime of the container.
Homepage Gitolite:
`The tests also use a somewhat convoluted system of environment variables in order to run entirely as a local user, without going through ssh at all. This lets a complete test suite run in about a lot less time than it would otherwise take.`

**Comment by garex** [comment]
@pumzie but to run em you anyway need some `apk add` commands. It will be good to have this here in some way.

**Comment by pumzie** [comment]
[bug.txt](https://github.com/ekapusta/docker-alpine-gitolite/files/612336/bug.txt)
Ok.
- Starting the container as described in Your Readme
- Log in the running container with `docker exec -it #ID /bin/sh`
- Now I'm logged in the running container with root.
- `adduser gittest`
- `su - gittest`
- `mkdir bin`
- `export PATH=~/bin:$PATH`
- `git clone git://github.com/sitaramc/gitolite`
- `cd gitolite`
- `GITOLITE_TEST=y prove`

To run a specific test:
`GITOLITE_TEST=y prove t/info-json.t`


update 24.11:
Using the same Version (2.1.4) then on my Debian don't make it better.

I added a file which shows the different behavior of Alpine and Debian.
I tracked down the first failure in the test.

**Comment by pumzie** [comment]
I got it :-).
It was the shebang in the generated hooks.
In `~/gitolite/src/triggers/repo-specific-hooks` there is a template for generating hooks.


```
__DATA__
#/bin/sh

# Determine what input the hook needs
# post-update takes args, pre/post-receive take stdin
```


The missing exclamation mark in `#/bin/sh` ist the problem in Alpine.

update 25.11.
The issue in Gitolite ist now fixed!
Tests are running fine.
